src/compute/src/sink/persist_sink.rs

```diff
@@ -127,9 +126,13 @@ where
                         data.swap(&mut buffer);
 
                         for ((key, value), ts, diff) in buffer.drain(..) {
-                            let key = key.unwrap_or_default();
-                            let value = value.unwrap_or_default();
-                            stash.entry(ts).or_default().push(((key, value), ts, diff));
+                            assert!(key.is_none(), "persist_sink does not support keys");
+                            let value = value.expect("persist_sink must have values");
+                            stash.entry(ts).or_default().push((
+                                (SourceData(Ok(value)), ()),
+                                ts,
+                                diff,
+                            ));
```

Comment:
This is the part that translates `(Option<Row>, Option<Row>)` records into `(SourceData, ())` records. It's probably wrong, but works at least for the example in the PR description.

Reply:
Actually, it looks like this might be correct after all. The `SinkRender` implement for `PersistSinkConnection` defines that persist sinks don't use keys: https://github.com/MaterializeInc/materialize/blob/main/src/compute/src/sink/persist_sink.rs#L35

... which in turn causes the dataflow rendering to package the whole row into the value: https://github.com/MaterializeInc/materialize/blob/main/src/compute/src/render/sinks.rs#L153

It looks like we only have the key-value separation for sinks because of the Kafka sink. Once we remove that from compute, much of this code should become simpler.